src/components/ui/BaseInput.vue

```diff
@@ -0,0 +1,97 @@
+<script setup lang="ts">
+import { ref, computed } from "vue";
+import OpenedEyePasswordIcon from "@/assets/images/eye-show-icon.svg?component";
+import ClosedEyePasswordIcon from "@/assets/images/eye-hidden-icon.svg?component";
+
+type inputType = "text" | "email" | "password";
+
+interface IProps {
+  inputType: inputType,
+  inputName: inputType,
+  label?: string,
+  error?: string,
+}
+
+const props = defineProps<IProps>();
+
+const inputClasses = computed(() => [`base-input__input--${props.inputName}`, {'base-input__input--is-invalid': !!props.error}]);
+
+const model = defineModel();
+
+const isPasswordHidden = ref(true);
+const fieldPassType = computed(() => props.inputName === "password" && isPasswordHidden.value ? "password" : "text");
+
+function showPassVisibility() {
+  isPasswordHidden.value = false;
+}
+
+function hidePassVisibility() {
+  isPasswordHidden.value = true;
+}
+</script>
+
+<template>
+  <div class="base-input">
+    <div v-if="props.label">{{ props.label }}</div>
+    <input
+      v-model="model"
+      :name="inputName"
+      :class="inputClasses"
+      :type="fieldPassType"
+    />
+
+    <OpenedEyePasswordIcon v-if="isPasswordHidden && props.inputName === 'password'" class="base-input__password-eye--opened" @click="showPassVisibility" />
+    <ClosedEyePasswordIcon v-else-if="props.inputName === 'password'" class="base-input__password-eye--closed" @click="hidePassVisibility" />
+
+    <span v-if="error" class="base-input__input--error">{{ error }}</span>
+  </div>
+  
+</template>
+
+<style scoped lang="scss">
+.base-input {
+  display: flex;
+  flex-direction: column;
+  position: relative;
+
+  &__input {
+    &--email, &--password {
```

Comment:
1) От --email нет смысла вообще
2) у тебя стили для --password вынесены ниже, поэтому:
```suggestion
    {
```